src/pages/Models/Details/ModelDetails.test.js

```diff
@@ -321,4 +321,26 @@ describe("ModelDetail Container", () => {
     const noRelationsMsg = wrapper.find("[data-testid='no-integrations-msg']");
     expect(noRelationsMsg.length).toBe(1);
   });
+
+  it("should show apps associated with a particular machine", () => {
+    const store = mockStore(dataDump);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[
+            "/models/pizza@external/hadoopspark?activeView=machines",
+          ]}
+        >
+          <QueryParamProvider ReactRouterRoute={Route}>
+            <TestRoute path="/models/*">
+              <ModelDetails />
+            </TestRoute>
+          </QueryParamProvider>
+        </MemoryRouter>
+      </Provider>
+    );
+
+    const machineApps = wrapper.find(".machine-app-icons img");
+    expect(machineApps.length).toBe(10);
```

Comment:
This test just checks that there are images on the page not that they are rendered appropriately per machine. Maybe spot check a few machines, the one with 4 images and one with 1?